sdk/cxx/src/address_mapper.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright 2017 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------
+*/
+#include "sawtooth/address_mapper.h"
+
+#include <exception>
+#include <iostream>
+#include <string>
+
+#include "cryptopp/sha.h"
+#include "cryptopp/filters.h"
+#include "cryptopp/hex.h"
+#include "log4cxx/logger.h"
+
+#include "proto/state_context.pb.h"
+
+namespace sawtooth {
+
+static log4cxx::LoggerPtr  logger(log4cxx::Logger::getLogger
+    ("sawtooth.address_mapper"));
+
+// Helper function to generate an SHA 512 hash and return it as a hex
+// encoded string.
+std::string SHA512(std::string message){
+    std::string digest;
+    CryptoPP::SHA512 hash;
+
+    CryptoPP::StringSource hasher(message, true,
+    new CryptoPP::HashFilter(hash,
+      new CryptoPP::HexEncoder (
+         new CryptoPP::StringSink(digest), false)));
```

Comment:
Seems expensive to create all these objects for each call. Could they be made persistent? Or are they cheap to create?